R/type-data-frame.R

```diff
@@ -89,6 +89,38 @@ df_ptype2 <- function(x, y, ..., x_arg = "", y_arg = "") {
   .Call(vctrs_df_ptype2, x, y, x_arg, y_arg)
 }
 
+# Fallback for data frame subclasses (#981)
+vec_ptype2_df_fallback <- function(x, y, x_arg = "", y_arg = "") {
+  ptype <- vec_ptype2(
+    new_data_frame(x),
+    new_data_frame(y)
+  )
+
+  if (!is_df_fallback(x) && !is_df_fallback(y)) {
+    types <- c(class(x)[[1]], class(y)[[1]])
+    msg <- cnd_type_message(x, y, x_arg, y_arg, NULL, "combine", NULL, types = types)
+    warn(c(
+      msg,
+      i = "vctrs coercion methods should be implemented for these classes.",
```

Comment:
I think it would be better to write this for the end user. Something like: "Convert all inputs to the same to avoid this warning"

Reply:
oh yeah I forgot to change this.